dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1208,249 +1255,391 @@ void TeachingTip::StartContractToClose()
-            {
-                rightBottomAvailable = false;
-                leftBottomAvailable = false;
-            }
-            // If the tip is too tall to fit between the bottom of the target and the bottom edge of the window.
-            if (tipHeight > static_cast<float>(windowBounds.Height) - (targetBounds.Height + targetBounds.Y))
-            {
+                break;
+            case winrt::TeachingTipHeroContentPlacementMode::Top:
                 bottomCenterAvailable = false;
                 bottomLeftAvailable = false;
                 bottomRightAvailable = false;
-            }
-
-            // If the tip is too wide to fit between the left edge of the target and the left edge of the window.
-            if (tipWidth > targetBounds.X)
-            {
-                leftCenterAvailable = false;
-                leftTopAvailable = false;
-                leftBottomAvailable = false;
-            }
-            // If the tip is too wide to fit between the center of the target and the left edge of the window.
-            if (contentWidth - MinimumTipEdgeToTailCenter() > targetBounds.X + (targetBounds.Width / 2.0f))
-            {
-                topLeftAvailable = false;
-                bottomLeftAvailable = false;
-            }
-            // If the tip is too wide to fit in the window when the tail is centered horizontally on the target and the tip.
-            if (contentWidth / 2.0f > targetBounds.X + targetBounds.Width / 2.0f ||
-                contentWidth / 2.0f > (windowBounds.Width - (targetBounds.Width + targetBounds.X) + (targetBounds.Width / 2.0f)))
-            {
-                topCenterAvailable = false;
-                bottomCenterAvailable = false;
-            }
-            // If the tip is too wide to fit between the center of the target and the right edge of the window.
-            if (contentWidth - MinimumTipEdgeToTailCenter() > windowBounds.Width - (targetBounds.X + (targetBounds.Width / 2.0f)))
-            {
-                topRightAvailable = false;
-                bottomRightAvailable = false;
-            }
-            // If the tip is too wide to fit between the right edge of the target and the right edge of the window.
-            if (tipWidth > static_cast<float>(windowBounds.Width) - (targetBounds.Width + targetBounds.X))
-            {
-                rightCenterAvailable = false;
-                rightTopAvailable = false;
                 rightBottomAvailable = false;
+                leftBottomAvailable = false;
+                break;
             }
+        }
 
+        // When ShouldConstrainToRootBounds is false screenSpaceAroundTarget = windowSpaceAroundTarget
+        // We have to separate them because there are checks which care about windowSpaceAroundTarget event when
+        // ShouldConstrainToRootBounds is true.
+        auto [targetBounds, windowSpaceAroundTarget, screenSpaceAroundTarget] = DetermineSpaceAroundTarget();
```

Comment:
>windowSpaceAroundTarget, screenSpaceAroundTarget [](start = 28, length = 48)

Think of better names

Reply:
I would recommend something like:
* clippedTargetBounds
* availableBoundsAroundTarget